experimental/sdk/webpubsub-graphql-subscribe/src/WpsWebSocketServer.ts

```diff
@@ -0,0 +1,112 @@
+import { WebPubSubServiceClient } from "@azure/web-pubsub";
+import { WebPubSubEventHandler } from "@azure/web-pubsub-express";
+import { ApolloServer} from "apollo-server-express";
+import { execute, GraphQLSchema, subscribe } from "graphql";
+import { SubscriptionServer } from 'subscriptions-transport-ws';
+import express from 'express';
+import WebSocket from "ws";
+import { WpsPubSub } from "./azure-wps-pubsub";
+import {config} from "./utils"
+
+/**
+  * a virtual WebSocket.Server without real server
+ */
+class VirtualWebSocketServer extends WebSocket.Server{
+	protocol: string = "graphql-ws";
+	readyState: number = WebSocket.OPEN;
+
+	constructor() {
+		 super({port: undefined, noServer: true}); 
+	}
+}
+
+/** 
+  * A VirtualWebSocketServer which is binded to an Azure Web PubSub client connection according to `servecClient` and `connectionId`
+  * It overrides method `send` and send all message to its binded connection in JSON format.
+ */
+class SubWpsWebSocketServer extends VirtualWebSocketServer{
+	serviceClient: WebPubSubServiceClient;
+	connectionId: string;
+
+	constructor(serviceClient: WebPubSubServiceClient, connectionId: string) {
+		super();
+		this.serviceClient = serviceClient;
+		this.connectionId = connectionId;
+	}
+
+	send(data: any) { this.serviceClient.sendToConnection(this.connectionId, data, { contentType: "text/plain" }); }
+}
+
+/** 
+  * A `VirtualWebSocketServer` which replaces original `WebSocket.Server` to communicate between the server and WebPub service using HTTP protocol
+  * `connectionId_to_ws` records the mapping from the `connectionId` of each client to its corrsponding `SubWpsWebSocketServer`.
+  * Finally `SubWpsWebSocketServer` sends messages to Web PubSub service.
+  * 
+ */
+class WpsWebSocketServer extends VirtualWebSocketServer {
+	app: any;
+	connectionId_to_ws: { [key:string]: SubWpsWebSocketServer } = {};
+	serviceClient: WebPubSubServiceClient;
+
+	constructor(wps_http_port: number, wps_conn_string: string, hub_name: string, express_server?: any){
+		super();
+		this.serviceClient = new WebPubSubServiceClient(wps_conn_string, hub_name);
+		this.app = express_server ? express_server : express();
+
+		let handler = new WebPubSubEventHandler(hub_name, ['*'], {
+			path: config.DEFAULT_WPS_MAIN_HANDLER_URL,
+			handleConnect: (req, res) => {
+				let connectionId = req.context.connectionId;
+				this.connectionId_to_ws[connectionId] = new SubWpsWebSocketServer(this.serviceClient, connectionId);
```

Comment:
class name: `SubWpsWebSocketServer`: how about `ClientConnectionContext`?

Reply:
How about Operation Context https://www.apollographql.com/docs/apollo-server/data/subscriptions/#operation-context ? we might need to reconstruct the `ConnectionContext`?

Reply:
If we replace Line 67 with codes below, logs will show that  `context` is still available and filled with content.
Does this prove that our package will not cause loss of "Operation Context".
```
	SubscriptionServer.create( { 
			schema, execute, subscribe, 	
			onConnect(connectionParams, webSocket, context) {
				console.log('Connected!')
				console.log(context);
			}, 
			onDisconnect(webSocket, context) {
				console.log('Disconnected!')
				console.log(context);
			},
		}, 
		createOptions,
	);
```